src/models/room-member.ts

```diff
@@ -163,7 +163,7 @@ export class RoomMember extends EventEmitter {
         const oldPowerLevel = this.powerLevel;
         const oldPowerLevelNorm = this.powerLevelNorm;
 
-        if (users[this.userId] !== undefined) {
+        if (users[this.userId] !== undefined && Number.isInteger(users[this.userId])) {
```

Comment:
I noticed we also have related but different checks in `room-state.ts` as part of `maySendEventOfType`, where `Number.isFinite` is used... Perhaps those should also require integers as well?